chore: Upgrade prisma to 6.7.0

zomars opened the pull request:
## What does this PR do?

Upgades Prisma to version 6.7.0.

Most of the changes are only imports:

```diff
-import type { Prisma } from "@prisma/client";
+import type { Prisma } from "@calcom/prisma/client";
```

Some notable changes are:

- Patches prismock library to keep unit testing passing with v6
- Fixes typings across next-auth files
- 

What do we get from upgrading?

Ability to join queries
Ability to get typed raw sql queries
It opens up the door for data residency with multiple clients
Allows us to to try the new TypeScript first `prisma-client` generator which should be faster for dev. To be done in a follow up.



## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Write details that help to start the tests -->

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

<!-- Remove bullet points below that don't apply to you -->

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

    
<!-- This is an auto-generated description by mrge. -->
---

## Summary by mrge
Upgraded Prisma and @prisma/client to version 6.7.0, updated the Prisma schema, and refactored user type usage for better type safety and compatibility.

- **Dependencies**
  - Updated all Prisma-related packages to 6.7.0.

- **Refactors**
  - Replaced `Prisma.UserArgs` with `Prisma.UserSelect` across the codebase.
  - Updated user type definitions and related code to match new Prisma patterns.
  - Added new types and interfaces for users and team members.
  - Updated Prisma schema output path and added primary key constraints.

<!-- End of auto-generated description by mrge. -->

zomars committed:
chore: Upgrade prisma to 6.7.0

zomars committed:
Build fixes

zomars committed:
type fixes

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
Update schema.prisma

zomars committed:
Patching

zomars committed:
Revert "Update schema.prisma"

This reverts commit 47d8618bf89ef4d007b30084df766f17281e21a1.

zomars committed:
Revert "Patching"

This reverts commit a1d2e3040e71690a44d4324db95d73b4d68c6adb.

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
Revert schema changes

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
WIP

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
Update getPublicEvent.ts

zomars committed:
Update imports

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
Update gitignore

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
update remaining imports

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
Delete .cursor/config.json

zomars committed:
Update _get.ts

zomars committed:
Discard changes to packages/features/eventtypes/lib/getPublicEvent.ts

zomars committed:
Update user.ts

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

Signed-off-by: Omar López <zomars@me.com>

# Conflicts:
#	apps/api/v2/src/ee/bookings/2024-08-13/controllers/e2e/user-bookings.e2e-spec.ts
#	packages/features/ee/workflows/components/WorkflowStepContainer.tsx
#	packages/lib/server/findUsersForAvailabilityCheck.ts

zomars committed:
Update .gitignore

zomars committed:
update

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
Update WorkflowStepContainer.tsx

zomars committed:
Update next-auth-custom-adapter.ts

zomars committed:
Update getPublicEvent.ts

zomars committed:
Update workflow.ts

zomars committed:
Update next-auth-custom-adapter.ts

zomars committed:
Update next-auth-options.ts

zomars committed:
Update bookingScenario.ts

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
fix missing imports

zomars committed:
upgrades prismock

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
patches prismock

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
Update reschedule.test.ts

zomars committed:
Update prisma.ts

zomars committed:
patch prismock

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
fix enums imports

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
Revert "Update prisma.ts"

This reverts commit 64edcf8db54171ff4456c209d563b5d431d99619.

zomars committed:
Revert "patch prismock"

This reverts commit e95819113dc9d88e7130947aa120cd42710977c8.

zomars committed:
fix patch

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Fix test that overrun the boundary, it shouldn't test too much

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Move prisma import to changeSMSLockState

emrysal committed:
Bring back broken test without illegal imports

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Merge with main and fix filter hosts by same round robin host

emrysal committed:
Fixed buildDryRunBooking fn tests

emrysal committed:
Fix and move ooo create or update handler test

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Fix packages/features/eventtypes/lib/isCurrentlyAvailable.test.ts

emrysal committed:
Fix packages/trpc/server/routers/viewer/organizations/listMembers.handler.test.ts

emrysal committed:
Mock @calcom/prisma

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Fix: verify-email.test.ts

emrysal committed:
fix: Moved WebhookService test and fixed default import mock

emrysal committed:
Fix: Added missing prisma mock, handleNewBooking uses that of course

emrysal committed:
We're not testing createContext here

emrysal committed:
fix: Prisma mock fix for listMembers.test.ts

emrysal committed:
More fixes to broken testcases

emrysal committed:
Forgot to remove borked test

emrysal committed:
Prevent the need to mock a lot of dependencies by moving out buildBaseWhereCondition to its own file

emrysal committed:
Temporarily skip getCalendarEvents, needs a rewrite

emrysal committed:
Fix: turns out you can access protected in testcases

emrysal committed:
fix further mocks

emrysal committed:
Added packages/features/insights/server/buildBaseWhereCondition.ts, types

emrysal committed:
Always great to have a mock and then not use it

emrysal committed:
And one less again.

emrysal committed:
fix: confirm.handler.test, didn't mock prisma

emrysal committed:
fix: Address minor nit by @eunjae & fix ImpersonationProvider test

emrysal committed:
Updated isPrismaAvailableCheck that doesn't crash on import

eunjae-lee commented on packages/eslint-plugin/src/rules/no-direct-prisma-import.ts:
```diff
@@ -0,0 +1,43 @@
+import { ESLintUtils } from "@typescript-eslint/utils";
```
🫶🏻

emrysal committed:
fix: Get Prisma directly from the client, it usually involves the Validator and does not need 'local' inclusion

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Add zod-prisma-types without the generator enabled (commented out)

emrysal committed:
Uncomment and see what happens

emrysal committed:
Change method of import as imports did not work in Input Schemas

emrysal committed:
Remove custom 'zod' booking model, it does not belong with Prisma

emrysal committed:
Fix all other global Model imports

emrysal committed:
Rewrite most schema includes AND remove barrel file

emrysal committed:
Add bookingCreateBodySchema to features/bookings

emrysal committed:
Flurry of type fixes for compatibility with new zod gen

emrysal committed:
Refactor out the custom prisma type createEventTypeInput

emrysal committed:
Work around nullable eventTypeLocations

emrysal committed:
HandlePayment type fix

emrysal committed:
More fixes, final fix remaining is CompleteEventType

emrysal committed:
Should fix a bunch more booking related type errors

emrysal committed:
Missed one

emrysal committed:
Some props missing from BookingCreateBodySchema

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Fix location type in handleChildrenEventTypes

emrysal committed:
Little bit hacky imo but it works

emrysal committed:
Final type error \o/

emrysal committed:
Forgot to include Prisma

emrysal committed:
Do not include zod-utils in booker/types

emrysal committed:
Oops, was already including Booker/types

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Fix membership type, also disallow updating createdAt/updatedAt, make part of patch/post

emrysal committed:
Fix api v1 type errors

emrysal committed:
Fix EventTypeDescription typings

emrysal committed:
Remove getParserWithGeneric, use userBodySchema with UserSchema

zomars commented on packages/features/bookings/lib/bookingCreateBodySchema.ts:
```diff
@@ -12,7 +12,7 @@ export const bookingCreateBodySchema = z.object({
   rescheduledBy: z.string().email({ message: "Invalid email" }).optional(),
   start: z.string(),
-  timeZone: timeZoneSchema,
+  timeZone: z.string().refine((value: string) => isSupportedTimeZone(value), { message: "Invalid timezone" }),
```
@emrysal was timeZoneSchema causing issues? I would like to keep a centralized schema for all timeZone validation.

zomars commented on packages/lib/checkRateLimitAndThrowError.ts:
```diff
@@ -48,6 +47,7 @@ export async function checkSMSRateLimit({
 }
 
 async function changeSMSLockState(identifier: string, status: SMSLockState) {
+  const prisma = (await import("@calcom/prisma")).prisma;
```
Is there a reason for this?

emrysal replied:
This was one of the first tests I fixed, will fix in a different way

zomars commented on packages/prisma/extensions/booking-idempotency-key.ts:
```diff
@@ -1,7 +1,8 @@
 import { v5 as uuidv5 } from "uuid";
 
-import { BookingStatus } from "@calcom/prisma/enums";
+import { Prisma } from "@calcom/prisma/client";
```
Should we use relative paths since we're inside the prisma package here? @emrysal

zomars commented on packages/prisma/schema.prisma:
```diff
@@ -10,13 +10,13 @@ datasource db {
-  output        = "./zod"
-  imports       = "./zod-utils"
-  relationModel = "default"
+  provider         = "zod-prisma-types"
```
@emrysal you're so brave for this

zomars replied:
I kinda get the move since "zod-prisma" it's been long abandoned. Blocking the upgrade. Kinda makes sense to do it now.

emrysal replied:
It was actually blocking the upgrade as the generated types did an import to the "old prisma" files that could not be fixed without forking; which I opted against.

emrysal committed:
use centralized timeZoneSchema

emrysal committed:
Implement feedback by @zomars

emrysal committed:
Couple of WIP pushes

joeauyeung committed:
Fix tests

joeauyeung commented on apps/api/v1/lib/validations/selected-calendar.ts:
```diff
@@ -10,14 +10,13 @@ export const schemaSelectedCalendarBaseBodyParams = SelectedCalendarSchema;
 
 export const schemaSelectedCalendarBodyParams = schemaSelectedCalendarBaseBodyParams
-  .partial({
+  .pick({
```
Instead of `omit` and constantly having to update this schema based on new fields, let's just pick the fields we need for the body based on the docs.

joeauyeung commented on apps/api/v1/test/lib/selected-calendars/_post.test.ts:
```diff
@@ -79,7 +79,7 @@ describe("POST /api/selected-calendars", () => {
         externalId: "ext123",
         userId: 444444,
-        id: "xxx-xxx",
+        id: "f47ac10b-58cc-4372-a567-0e02b2c3d479",
```
Zod validation included `.uuid()` so we need to pass an actual uuid for this to pass.

joeauyeung committed:
Type fixes in `handleChildrenEventTypes` test

emrysal committed:
Try and parse metadata before use

emrysal committed:
Merge branch 'upgrade/prisma-6.7.0' of https://github.com/calcom/cal.com into upgrade/prisma-6.7.0

emrysal committed:
Change zod-prisma-types configuration for optimal performance

joeauyeung committed:
Fix prisma validator error in `prisma/selects/credential`

emrysal committed:
Devin didn't get the memo that we were picking instead of omit

emrysal committed:
Merge branch 'upgrade/prisma-6.7.0' of https://github.com/calcom/cal.com into upgrade/prisma-6.7.0

emrysal committed:
Disable seperate relations model, hits a bug

emrysal committed:
Import absolute - this makes rollup work in @platform/libraries

emrysal committed:
Attempt at removing resolutions override

joeauyeung committed:
Refactor using `Prisma.validator` to `satisfies`

joeauyeung committed:
Merge branch 'upgrade/prisma-6.7.0' of https://github.com/calcom/cal.com into upgrade/prisma-6.7.0

emrysal committed:
Build atoms using @calcom/prisma/client

emrysal committed:
Build atoms using @calcom/prisma/client

zomars committed:
fixes

zomars committed:
Update eventTypeSelect.ts

zomars commented on packages/app-store/_utils/stripe.ts:
```diff
@@ -24,14 +24,12 @@ export async function getStripeCustomerIdFromUserId(userId: number) {
   return customerId;
 }
 
-const userType = Prisma.validator<Prisma.UserArgs>()({
```
I'm curious. What trouble is the select validator causing @joeauyeung ?

joeauyeung replied:
This was throwing a ton of `prisma.client.validator` is undefined errors in the tests. I think this change is good as well because Prisma recommends using typescript's `satisfies` to reduce overhead.

https://www.prisma.io/blog/satisfies-operator-ur8ys8ccq7zb

joeauyeung committed:
Adjust `eventTypeMetaDataSchemaWithUntypedApps` from `unknown` to `record(any)`

joeauyeung committed:
Merge branch 'upgrade/prisma-6.7.0' of https://github.com/calcom/cal.com into upgrade/prisma-6.7.0

joeauyeung committed:
`EventTypeDescription` rely on `descriptionAsSafeHTML` instead of `description`

joeauyeung committed:
Add `seatsPerTimeSlot` to event type public select

joeauyeung committed:
Fix typing in `users-public-view` getServerSide props

joeauyeung committed:
Add missing `schedulingType` to prop

supalarry committed:
chore: bump platform libraries

emrysal committed:
Function return type is illegal, not sure how this passed eslint (#21567)

emrysal committed:
Merged with main

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
Update updateTokenObject.ts

zomars committed:
Update handleResponse.ts

zomars committed:
Update index.ts

zomars committed:
Update handleChildrenEventTypes.ts

zomars committed:
Update booking-idempotency-key.ts

zomars committed:
Update WebhookService.test.ts

zomars committed:
Update events.test.ts

zomars committed:
Update queued-response.test.ts

zomars committed:
Update events.test.ts

zomars committed:
Update getRoutedUrl.test.ts

zomars committed:
fix: type checks

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
fixes

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
chore: bump platform libraries

zomars committed:
Update yarn.lock

zomars committed:
more fixes

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
fixes

Signed-off-by: Omar López <zomars@me.com>

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
biuld fixes

zomars committed:
chore: bump platform libraries

zomars committed:
Update conferencing.repository.ts

zomars committed:
Update conferencing.repository.ts

zomars committed:
Update getCalendarsEvents.test.ts

zomars committed:
Update vite.config.js

zomars committed:
chore: bump platform libraries

zomars committed:
Update users.ts

ThyMinimalDev commented:
will try to patch platform-libraries-0.0.2 in order to use the new location for prisma client

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
Discard changes to docs/api-reference/v2/openapi.json

zomars committed:
Update vite.config.ts

zomars committed:
updated platform libraries

zomars committed:
Update get.handler.test.ts

zomars committed:
Update get.handler.test.ts

zomars committed:
Update schema.prisma

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
Discard changes to docs/api-reference/v2/openapi.json

zomars committed:
Update next-auth-custom-adapter.ts

zomars committed:
Update team.ts

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Flurry of type fixes

emrysal committed:
Fix majority of insight related type errors

emrysal committed:
Type fixes for unlink of account

emrysal committed:
Make user nullable again

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Fixed a bunch of unit tests and one type error

emrysal committed:
Attempted mock fix

emrysal committed:
Attempted fix for Attribute type

emrysal committed:
Ensure default import becomes prisma, but not direct usage

emrysal committed:
Import default as prisma in prisma.module

emrysal committed:
Add attributeOption to attribute type

emrysal committed:
Fix calcom/prisma mock

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

zomars committed:
Refactor Prisma client imports to @calcom/prisma/client

Updated all imports from '@prisma/client' to '@calcom/prisma/client' across tests and repository files for consistency and to use the correct Prisma client package. This change improves maintainability and ensures the correct client is referenced throughout the codebase.

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Undo removal of max-warnings=0 to get main to merge

emrysal committed:
Remove unit tests for e2e fixtures, provide new prisma mock

emrysal committed:
Mock @calcom/prisma in event manager

emrysal committed:
Mock @calcom/prisma in event manager

emrysal committed:
Add correct format even with --no-verify

emrysal committed:
Mock prisma in CalendarManager

emrysal committed:
Add mock for permission-check.service

emrysal committed:
Better injection in PrismaApiKeyRepository imports

emrysal committed:
More mock fixes :)

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Fix listMembers.handler.test

emrysal committed:
Merge branch 'upgrade/prisma-6.7.0' of https://github.com/calcom/cal.com into upgrade/prisma-6.7.0

emrysal committed:
Fix User import

emrysal committed:
Appropriately adjust all types to be imported as types, there were a lot of types imported as normal deps

emrysal committed:
Why was this a thing?

emrysal committed:
Strictly speaking; Not using prismock anymore

emrysal committed:
Ditched patch file for prismock

emrysal committed:
Fix output.service.ts platform type imports, need concrete for plainToClass

emrysal committed:
Better typing and tests for unlinkConnectedAccount.handler

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Small type fix

emrysal committed:
Disable calendar cache tests as they are dependent on prismock

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

ThyMinimalDev committed:
chore: bump platform lib

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
getRoutedUrl test remove of unused import

emrysal committed:
Extract select to external const on getEventTypesFromDB

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Direct select of userSelect from selects/user

emrysal committed:
fix type error from merging 23653

emrysal committed:
Fixed integration tests by removing hardcoded values that were possible due to mocking, but as its now directly hitting the db no longer

ThyMinimalDev commented on packages/platform/examples/base/src/lib/prismaClient.ts:
```diff
@@ -1,4 +1,4 @@
-import { PrismaClient } from "@prisma/client";
+import { PrismaClient } from "@calcom/prisma/client";
```
and its own prisma client, it is not communicating with cal.com db

ThyMinimalDev committed:
fix: vite config atoms prisma client type location

ThyMinimalDev committed:
revert: example app prisma client

ThyMinimalDev committed:
revert: example app prisma client

ThyMinimalDev committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

ThyMinimalDev committed:
bump platform libs

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

ThyMinimalDev committed:
fix: use class instead of type for DI of PlatformBookingsService

hbjORbj committed:
Merge branch 'main' into upgrade/prisma-6.7.0

ThyMinimalDev committed:
update platform libs

ThyMinimalDev committed:
remove unused variable

ThyMinimalDev committed:
chore: generate prisma client for api v2

ThyMinimalDev committed:
fix: api v2 e2e

ThyMinimalDev committed:
fix: atoms e2e

ThyMinimalDev committed:
fix: atoms e2e

ThyMinimalDev committed:
fix: atoms e2e

ThyMinimalDev committed:
fix: api v2 e2e

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

ThyMinimalDev committed:
fix: tsconfig apiv2 enums

ThyMinimalDev committed:
publish libraries

emrysal committed:
Merge branch 'main' into upgrade/prisma-6.7.0

emrysal committed:
Simplify check for existence teamId

emrysal reviewed (approved):
LGTM 🚀 - huge.
